### PR TITLE
WIP: pkg/operator/quorumguardcontroller: add pod affinity

### PR DESCRIPTION
**Pull request by hexfusion**
Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

### CODE REVIEW

**Comment by hexfusion** [comment]
cc @hasbro17

**Comment by hexfusion** [comment]
/hold 

I think nodeSelector doesn't work here as it circumvents the scheduler.

**Comment by hasbro17** [comment]
/retitle WIP: pkg/operator/quorumguardcontroller: add pod affinity

File: pkg/operator/quorumguardcontroller/quorumguardcontroller.go [hunk 1]
```diff
@@ -172,7 +173,45 @@ func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, r
...
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "k8s-app",
+								Operator: metav1.LabelSelectorOpIn,
+								Values:   []string{"etcd-quorum-guard"},
+							},
+						},
+					},
+					TopologyKey: "kubernetes.io/hostname",
```
Comment (hasbro17) [inline pkg/operator/quorumguardcontroller/quorumguardcontroller.go:190]: You're probably reworking this but this is already present here:
https://github.com/openshift/cluster-etcd-operator/blob/83e4a83079cfffbd3d8048f02f6d0603cd00c15a/pkg/operator/etcd_assets/bindata.go#L966-L973

**Comment by hexfusion** [comment]
going to raise a new PR as this has changed dramatically

**Pull request closed without merging**